code/modules/mob/living/simple_animal/hostile/giant_spider.dm

```diff
@@ -480,12 +482,24 @@
 	desc = "Set a directive for your children to follow."
 	check_flags = AB_CHECK_CONSCIOUS
 	button_icon_state = "directive"
+	
+/datum/action/innate/spider/set_directive/IsAvailable()
+	if(..())
+		if(!istype(owner, /mob/living/simple_animal/hostile/poison/giant_spider))
+			return FALSE
+		var/mob/living/simple_animal/hostile/poison/giant_spider/S = owner
+		if(S.playable_spider)
+			return FALSE
+		return TRUE
 
 /datum/action/innate/spider/set_directive/Activate()
 	if(!istype(owner, /mob/living/simple_animal/hostile/poison/giant_spider/nurse))
 		return
 	var/mob/living/simple_animal/hostile/poison/giant_spider/nurse/S = owner
-	S.directive = stripped_input(S, "Enter the new directive", "Create directive", "[S.directive]")
+	if(!S.playable_spider)

```

Comment:
This variable is only true if the spider has been created by a sentient nurse spider. It's the same variable that determines wether or ghosts can join as the spider.

Reply:
is that based on inher from `IsAvailable` or something? I'm looking specifically for the code that determines "generation".

Reply:
https://github.com/tgstation/tgstation/blob/master/code/modules/mob/living/simple_animal/hostile/giant_spider.dm line 467 and https://github.com/tgstation/tgstation/blob/master/code/game/objects/effects/spiders.dm line 196

Reply:
Oh so playablespider is NOT given to pots/mindswappers but normal reproduction is.

How confusing that even if you are a playable spider you may not have the playable spider var.